third_party/move/move-vm/runtime/src/loader/mod.rs

```diff
@@ -140,6 +169,7 @@ pub(crate) struct Loader {
     module_cache: RwLock<ModuleCache>,
     type_cache: RwLock<TypeCache>,
     natives: NativeFunctions,
+    name_cache: RwLock<StructNameCache>,
```

Comment:
How about moving the RwLock inside `StuctNameCache` like `data: RwLock<(TheMap, TheVector)>`

Then you can have more efficient way inside `insert_or_get`